lib/core_ext/numpy.rb

```diff
@@ -59,6 +73,44 @@ def roll(shift, axis: 0)
     end
   end
 
+  def assign(x1, y1, x2, y2, value)
+
+    x1.upto(x2) { |x| y1.upto(y2) { |y| self[y][x] = value } }
+
+    self
+  end
+
+  def paste(x, y, other)
+    raise ArgumentError, 'must be in bounds' if x < 0 || y < 0 || x + other.first.length >= first.length || y + other.length >= length
+
+    other.each.with_index { |row, y2| row.each.with_index { |_val, x2| self[y2 + y][x2 + x] = other[y2][x2] } }
+
+    self
+  end
+
+  def adjust(x1, y1, x2, y2, difference)
+    pts!(x1, y1, x2, y2)
+
+    x1.upto(x2) { |x| y1.upto(y2) { |y| self[y][x] += difference } }
+
+    self
+  end
+
+  def clip(x1, y1, x2, y2)
+    pts!(x1, y1, x2, y2)
+
+    new_array = Array.new(y2 - y1 + 1).map { Array.new(x2 - x1 + 1) }
+    x1.upto(x2) { |x| y1.upto(y2) { |y| new_array[y - y1][x - x1] = self[y][x] } }
+
+    new_array
+  end
+
+  def invert(x1, y1, x2, y2)
+    pts!(x1, y1, x2, y2)
+
+    x1.upto(x2) { |x| y1.upto(y2) { |y| self[y][x] = self[y][x] == 0 ? 1 : 0 } }
+  end
```

Comment:
this seems really specific, do we need it? If so, I would probably call this `not` since it doesn't invert. Actually, numpy has an `invert` method that does a bitwise invert. This my confusion... If you want to keep it, add a comment pls too